src/components/Ingredients/IngredientForm.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { CONFIRM_ICON, TRASH_ICON } from '../../helpers/icons';
+import { editIngredient, addIngredient } from '../../store/recipes/actions';
+import { v4 as uuidv4 } from 'uuid';
+import { IIngredient } from '../../store/interfaces';
+
+const IngredientForm = ({
+  recipeId,
+  ingredientId,
+  amount,
+  name,
+  unit,
+  isIngredientInEdition, //false=empty; true=filled
+  isNewIngredientAdded,
+  closeIngredientEdition,
+  deleteCurrentIngredient,
+  deleteNewIngredient,
+}: {
+  recipeId: number | string;
```

Comment:
Widzę, że w wielu miejscach przekazywany jest **typ id przepisu** - `number | string`.
1. Przyszłościowo byłoby dobrze, gdyby ten identyikator był jednoznacznie albo numerem, albo stringiem. 
2. I teraz, jeśli będziesz chciała to ujednolicić w przyszłości, trzeba będzie odszukać te wszystkie wystąpienia z osobna. Jako że każde z nich odnosi się do tego samego "elementu", czyli pola id w interfejsie IRecipe, - ułatwieniem w takim przypadku mogłoby być otypowanie ich w odniesieniu do tego interfejsu: `IRecipe['id']`. Wtedy wystarczyłoby zmienić to w jednym miejscu: w interfejsie 🙂 

_Mała dygresja: w typescripcie fajne jest właśnie to, że nie trzeba typować wszystkiego i wszędzie "ręcznie" - można korzystać z istniejących interfejsów (jak wyżej), można je rozszerzać (Extends), wybierać część właściwości interfejsu (Pick, Omit, Partial), no i jest też inferencja typów, która sprawia, że TS "domyśla się" typu zmiennych na podstawie deklaracji na wyższym poziomie._